api/src/convert/amount.rs

```diff
@@ -119,3 +121,21 @@ impl TryFrom<&external::Receipt_oneof_masked_amount> for MaskedAmount {
         }
     }
 }
+
+impl From<&Amount> for external::Amount {
+    fn from(source: &Amount) -> Self {
+        let mut amount = external::Amount::new();
+        amount.set_value(source.value);
+        amount.set_token_id(*source.token_id);
+        amount
+    }
+}
+
+impl From<&external::Amount> for Amount {
+    fn from(source: &external::Amount) -> Self {
+        Amount {
+            value: source.get_value(),
+            token_id: TokenId::from(source.get_token_id()),
+        }
```

Comment:
Optional/Equivalent: Use `Amount::new` and `into()`
```suggestion
        Amount::new(source.get_value(), source.get_token_id().into())
```